src/data/data.ts

```diff
@@ -752,6 +756,53 @@ export const getExchangeRates = (
   );
 };
 
+/**
+ */
+export const getPublicKeyForUsername = (
+  username: string,
+  options?: RequestOptions
+): Promise<string> => {
+  const endpoint = 'api/v0/get-public-key-for-user-name';
+  return api.get(
+    (options?.nodeURI ? cleanURL(options.nodeURI, endpoint) : endpoint) + `/${username}`
+  );
+};
+
+/**
+ */
+export const getUsernameForPublicKey = (
+  publicKey: string,
+  options?: RequestOptions
+): Promise<string> => {
+  const endpoint = 'api/v0/get-user-name-for-public-key';
+  return api.get(
+    (options?.nodeURI ? cleanURL(options.nodeURI, endpoint) : endpoint) + `/${publicKey}`
+  );
+};
+
+export const getTransactionInfo = (
+  params: Partial<APITransactionInfoRequest> = {},
```

Comment:
Is it valid to call this without passing any params? If there are some required fields you could use the `PartialWithRequiredFields` type.

Reply:
Ahh yeah - you're right. It's a little annoying since it's a little bit of either/or. You specify this field OR that one. I'll try to clean up in the AM. Thanks!

Reply:
ahh, yeah if it helps there's a few examples of where I dealt with those via the union operator thing https://github.com/deso-protocol/deso-js/blob/3459fec2682a1fed4c57e39075972d2c1c31842b/src/data/data.ts#L193